store/src/db/sqlite.rs

```diff
@@ -0,0 +1,35 @@
+use crate::db::interface::Db;
+use crate::types::BlockNumber;
+use anyhow::anyhow;
+use deadpool_sqlite::{rusqlite::types::ValueRef, Pool};
+use miden_crypto::hash::rpo::RpoDigest;
+use miden_crypto::utils::{Deserializable, SliceReader};
+
+#[tonic::async_trait]
+impl Db for Pool {
```

Comment:
Note: Implementing the trait on the pool has some tradeoffs

- Implementing it on the connection itself would make the user code non-async. The idea here was to make the trait async to support changing the underlying DB with minimal changes
- This could be implemented on a wrapper type, or using generics instead. That would maintain the future proofness of the code, but would make it slower to compile and harder to understand. But that is a viable option if we don't want to have the deadpool if we change the DB in the future (I think it should be fine though)